src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -170,25 +177,93 @@ private void loadData(Executor executor) throws IllegalStateException {
+        // skip: EST: -05:00; HST: -10:00; MST: -07:00
+        // kernel will handle EST, HST, MST
+        // ZoneId.SHORT_IDS is deprecated, so it will not probably change
+      } else {
+        Integer index = zoneIdToTable.get(mapTo);
+        // some short IDs are DST, skip unsupported
+        if (index != null) {
+          data.add(new HostColumnVector.StructData(id, index));
+        }
+      }
+    }
+    shortIDs = HostColumnVector.fromStructs(type, data);
+  }
+
+  public ColumnVector getTimeZoneShortIDs() {
+    return shortIDs.copyToDevice();
+  }
+
   @SuppressWarnings("unchecked")
   private void doLoadData() {
     synchronized (this) {
       try {
         Map<String, Integer> zoneIdToTable = new HashMap<>();
         List<List<HostColumnVector.StructData>> masterTransitions = new ArrayList<>();
+        // Build a timezone ID index for the rendering of timezone IDs which may be included in datetime-like strings.
+        // For instance: "2023-11-5T03:04:55.1 Asia/Shanghai" -> This index helps to find the
+        // offset of "Asia/Shanghai" in timezoneDB.
+        //
+        // Currently, we do NOT support all timezone IDs. For unsupported time zones, like invalid ones,
+        // we replace them with NULL value when ANSI mode is off when parsing string to timestamp.
+        // This list only contains supported time zones.
+        List<String> zondIdList = new ArrayList<>();
+        List<String> unsupportedZoneList = new ArrayList<>();
+        
+        // collect zone id and sort
+        List<ZoneId> ids = new ArrayList<>();
         for (String tzId : TimeZone.getAvailableIDs()) {
           ZoneId zoneId;
           try {
             zoneId = ZoneId.of(tzId).normalized(); // we use the normalized form to dedupe
+            ids.add(zoneId);
           } catch (ZoneRulesException e) {
             // Sometimes the list of getAvailableIDs() is one of the 3-letter abbreviations, however,
             // this use is deprecated due to ambiguity reasons (same abbrevation can be used for 
             // multiple time zones). These are not supported by ZoneId.of(...) directly here.
             continue;
           }
+        }
+        Collections.sort(ids, new Comparator<ZoneId>() {
+          @Override
+          public int compare(ZoneId o1, ZoneId o2) {
+            // sort by `getId`
+            return o1.getId().compareTo(o2.getId());
+          }
+        });
+
+        // A simple approach to transform LocalDateTime to a value which is proportional to
+        // the exact EpochSecond. After caching these values along with EpochSeconds, we
+        // can easily search out which time zone transition rule we should apply according
+        // to LocalDateTime structs. The searching procedure is same as the binary search with
+        // exact EpochSeconds(convert_timestamp_tz_functor), except using "loose instant"
+        // as search index instead of exact EpochSeconds.
```

Comment:
> So what is the difference in storing the data as a looseEpochSeconds vs just storing the before/after time related to the gap as epoch seconds? 

Yes, epoch second is better, it will save the computation of looseEpochSeconds.

> It also simplifies using the column in the future.

Using structure (year, month, day ...) column is also an option.

Will select one of above solutions. Currently let's use looseEpochSeconds.
I'll consider with DST together.